corehq/apps/users/util.py

```diff
@@ -109,6 +109,31 @@ def username_to_user_id(username):
     return user._id
 
 
+def cached_location_id_to_name(location_id):
+    if not location_id:
+        return None
+
+    key = 'location_id_username_cache_{id}'.format(id=location_id)
+    ret = cache.get(key)
+    if ret:
+        return ret
+    else:
+        ret = location_id_to_name(location_id)
+        cache.set(key, ret)
```

Comment:
nit: I was thinking we should set this only if a name was returned.
but we could get missing location ids when locations are removed though, so this seems okay as well.